lib/KeyWrapper.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+# include "Libconfig.h"
+# include <cstring>
+# include <memory>
+
+namespace LibConfig
+{
+  struct KeyWrapper
```

Comment:
I'd expect move assignment operators and a move constructor.
Since you have a method to copy the key, it seems odd to have the copy ones.